src/UKI.jl

```diff
@@ -0,0 +1,242 @@
+module UKI
+
+using LinearAlgebra
+using DocStringExtensions
+
+export UKIObj
+export update_ensemble!
+"""
+    KIObj{FT<:AbstractFloat, IT<:Int}
+
+Struct that is used in Unscented Kalman Inversion (UKI)
+
+#Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct UKIObj{FT<:AbstractFloat, IT<:Int}
+    "vector of parameter names (never used)"
+     unames::Vector{String}
+    "a vector of arrays of size N_ensemble x N_parameters containing the mean of the parameters (in each UKI iteration a new array of mean is added)"
+     θ_bar::Vector{Array{FT}}
+     "a vector of arrays of size N_ensemble x (N_parameters x N_parameters) containing the covariance of the parameters (in each UKI iteration a new array of cov is added)"
+     θθ_cov::Vector{Array{FT, 2}}
+     "vector of observations (length: N_data); mean of all observation samples"
+     g_t::Vector{FT}
+     "covariance of the observational noise"
+     obs_cov::Array{FT, 2}
+     "a vector of arrays of size N_ensemble x N_g containing the predicted observation (in each UKI iteration a new array of predicted observation is added)"
+     g_bar::Vector{Array{FT}}
+     "ensemble size"
+     N_ens::IT
+     "parameter size g = G(θ)"
+     N_θ::IT
+     "observation size g = G(θ)"
+     N_g::IT
+     "weights in UKI"
+     sample_weights::Array{FT}
+     μ_weights::Array{FT} 
+     cov_weights::Array{FT} 
+     "regularization parameter"
+     α_reg::Float64
+end
+
+
+
+"""
+UKIObj Constructor 
+parameter_names::Vector{String} : parameter name vector
+θ0_bar::Array{FT} : prior mean
+θθ0_cov::Array{FT, 2} : prior covariance
+g_t::Vector{FT} : observation 
+obs_cov::Array{FT, 2} : observation covariance
+α_reg::Float64 : regularization parameter toward θ0 (0 < α_reg <= 1), default should be 1, without regulariazion
+"""
+function UKIObj(parameter_names::Vector{String},
+                θ0_bar::Array{FT}, 
+                θθ0_cov::Array{FT, 2},
+                g_t::Vector{FT}, # observation
```

Comment:
Similarly, it would be better if we have access to a constructor similar to `EKI.jl` (not saying this one should be deleted).